UnitySetup/UnitySetup.psm1

```diff
@@ -1870,7 +1870,25 @@ function Start-UnityEditor {
             $process.Start() | Out-Null
 
             if ( $Wait ) {
-                $process.WaitForExit()
+                if( $LogFile ) {
+                    while ( -not (Test-Path $LogFile -Type Leaf) ) {
+                        Start-Sleep -Milliseconds 100
+                    }
+
+                    $ljob = Start-Job -ScriptBlock { param($log) Get-Content "$log" -Wait } -ArgumentList $LogFile
+
+                    while ( -not $process.HasExited )
+                    {
+                        Receive-Job $ljob
+                    }
+
+                    Receive-Job $ljob
```

Comment:
Still think you should loop on this post process exited Receive-Job call while the job has data - I bet in 99% of the cases that'll get the whole log file.

Reply:
Not sure what you mean or how to do that lol. Any advice?

Reply:
```
while($ljob.HasMoreData) { Receive-Job $ljob }
```

Reply:
If the file never gets created HasMoreData should be false so we should be safe to check for more data. The only race here is that we'll loop before the job gets a chance to process and retrieve more data.

Reply:
This still won't return though. I've tried this but my task never completes

Reply:
Really? The job always has more data? That's strange indeed.

Reply:
Yeah I think it's bc there's some other process still running from Unity that's attached itself to the log file somehow. I really can't tell ya what's going on tbh.

I noticed I also saw a speed up in the task completion by switching to this as well. Before when I had the `-Wait` flag it would finish a lot longer than when it printed out in the console it was gonna exit. I always thought it was bc of the compiler process that was running in the background that Unity uses. idk. But we don't need to wait for that process to end, cause we just care about the editor itself.

Reply:
Actually I see another issue - all this output from the jobs is getting pushed to the output stream of Start-UnityEditor. I'm thinking it should probably get passed directly to Write-Host so that it doesn't get muddled with the normal output of the cmdlet.

For the HasMoreData issue, you could cache the results of Receive-Job and exit if it didn't actually give you anything:

Something like: 
```
do {
  $logOutput = Receive-Job
  $logOutput | Write-Host
} until($logOutput.Length == 0)
```

Reply:
I'm using `-PassThru` anyway, not sure if that'll change anything. 

https://github.com/microsoft/unitysetup.powershell/pull/201#discussion_r413373793